### ISSUE TITLE
Ability to disable JavaScript for a browsing context

### ISSUE THREAD

**Issue by OrKoN** [w3c/webdriver-bidi#746]
Chrome allows disabling JavaScript for a specific page: https://pptr.dev/api/puppeteer.page.setjavascriptenabled/

Does it make sense to have this functionality in WebDriver BiDi?

**Comment by jgraham** [comment]
Yes, I think this makes sense. I'm not 100% sure if Gecko can disable it per (top level) navigable, but in any case it would make sense to have a command that either allowed disabling globally or for certain top-level navigables.

**Comment by whimboo** [comment]
> I'm not 100% sure if Gecko can disable it per (top level) navigable

The BrowsingContext webidl has a `allowJavascript`

**Comment by shs96c** [comment]
What should happen when JS is disabled yet a session wants to evaluate some JS? I believe that should still work, but it's best to confirm

**Comment by OrKoN** [comment]
@shs96c yeah, that should still work: only the scripts coming from the page should not be executed as if JS is not supported.

**Comment by juliandescottes** [comment]
> The BrowsingContext webidl has a [`allowJavascript` flag](https://searchfox.org/mozilla-central/rev/f3e4b33a6122ce63bf81ae8c30cc5ac37458864b/dom/chrome-webidl/BrowsingContext.webidl#236-244) which would exactly allow us to do that on a per top-level browsing context level.

For info that's what we use for FF DevTools' "Disable JavaScript" feature 

<img width="154" alt="image" src="https://github.com/user-attachments/assets/1667e1ea-824a-4243-82aa-c0ac046381da">

**Comment by sadym-chromium** [comment]
The CDP method is [`Emulation.setScriptExecutionDisabled`](https://chromedevtools.github.io/devtools-protocol/tot/Emulation/#method-setScriptExecutionDisabled).

**Comment by OrKoN** [comment]
Note that Chrome still allows script evaluation via CDP if JS is disabled for the page and clients like Puppeteer expect this ([test](https://github.com/puppeteer/puppeteer/blob/main/test/src/page.spec.ts#L2044)).

**Comment by jgraham** [comment]
Note that as specified today, disabling scripting in the browser (or a specific browsing context) will disable injected scripts. That's because we call https://html.spec.whatwg.org/#creating-a-classic-script which sets the source to an empty string in the case that scripting is disabled.

**Comment by sadym-chromium** [comment]
Spec draft: https://github.com/w3c/webdriver-bidi/pull/946

**Comment by sadym-chromium** [comment]
Spec is landed in https://github.com/w3c/webdriver-bidi/pull/946
WPT are in https://github.com/web-platform-tests/wpt/pull/54289

**Comment by whimboo** [comment]
This is all done.

**Comment by sadym-chromium** [comment]
HTML spec is still not landed, so reopening. https://github.com/whatwg/html/pull/11441

**Comment by whimboo** [comment]
HTML spec was merged on Dec 12, 2025
